model/vsc_parser.py

```diff
@@ -34,93 +34,151 @@
-# found near the bottom.
 
 # Base class for all nodes in the tree
 class AST(anytree.Node):
-   def __init__(self, name, parent):
-       super().__init__(name, parent)
-
-class Argument(AST):  # for in_arguments and out_arguments
-   name: str
-   description: str
-   type: str
-
-class Method(AST):
-   name: str
-   description: str
-   in_arguments: list[Argument]
-   out_arguments: list[Argument]
-
-class Event(AST):  # Note: Event details are TBC
-   name: str
-   description: str
-   in_arguments: list[Argument]
-
-class Property(AST):
-   name: str
-   description: str
-   type: str
-   datatype:str
-
-class Member(AST):
-   name: str
-   datatype: str
-   description: str
-
-class Option(AST):
-   name: str
-   value: str
-
-class Struct(AST):
-   name: str
-   description: str
-   members: list[Member]
-
-
-class Typedef(AST):
-   name: str
-   datatype: str
-   description: str
-   min: str
-   max: str
-
-class Enum(AST):
-   name: str
-   datatype: str
-   description: str
-   options: list[Option]
-   members: list[Member]
-
-class Namespace(AST):
-   name: str
-   description: str
-   structs: list[Struct]
-   typedefs: list[Typedef]
-   enums: list[Enum]
-   methods: list[Method]
-   events: list[Event]
-   properties: list[Property]
-
-class Service(AST):
-   name: str
-   description: str
-   namespaces: list[Namespace]
+    def __init__(self, name, parent):
+        super().__init__(name, parent)
+
+# AST node types.  They are all subtypes of the AST node.  They are empty
+# because the member variables are later added at run-time, as defined by
+# the schema variable.
+class Argument(AST): pass
+class Enum(AST): pass
+class Error(AST): pass
+class Event(AST):  pass
+class Include(AST): pass
+class Member(AST): pass
+class Method(AST): pass
+class Namespace(AST): pass
+class Option(AST): pass
+class Property(AST): pass
+class Service(AST): pass
+class Struct(AST): pass
+class Typedef(AST): pass
+
+# Some identifiers used to indicate optionality
+MAN = 0 # MAN(datory)
+REC = 1 # REC(ommended)
+OPT = 2 # OPT(ional)
+
+schema = {
+    # Root node can theoretically contain several services
+    # ... one Service = one file, presumably.
+    "AST" : {
+        "services" : (list[Service], MAN)
+        },
+
+    "Argument" : {
+        "name" : (str, MAN),
+        "description" : (str, REC),
+        "datatype" : (str, MAN),
+        "arraysize" : (str, OPT),
+        "range"  : (str, OPT)
+        },
+
+    "Error" : {
+        "datatype" : (str, MAN),
+        "arraysize" : (str, OPT),
+        "range" : (str, OPT)
+        },
+
+    "Method" : {
+        "name" : (str, MAN),
+        "description" : (str, MAN),
+        "in" : (list[Argument], OPT),
+        "out" : (list[Argument], OPT),
+        "errors" : (Error, REC)
+        },
+
+    "Event" : {
+        "name" : (str, MAN),
+        "description" : (str, REC),
+        "in" : (list[Argument], OPT)
+        },
+
+    "Property" : {
```

Comment:
See comment in other VSC PR. Is it aligned that properties always are read-write, i.e. no need to include any information concerning that? (The current VSC seat example include a field `type` that specifies `attribute`, likely inspired by VSS.

```
properties:
      - name: a_property # Will this be acceptable?
        description: A signal
        type: attribute # or sensor.
        datatype: uint8 # Native datatypes only?
```

Reply:
> Is it aligned that properties always are read-write, i.e. no need to include any information concerning that? 

I don't know but it's a question for the VSC definition itself. (Can you create an issue for this question?)
  
 > (The current VSC seat example include a field type that specifies attribute, likely inspired by VSS.

Yes I noticed the difference compared to the previous parser but decided to implement following the README ([Properties](https://github.com/COVESA/vehicle_service_catalog#namespace-list-object-properties)) as closely as I could, since it acts as our specification at the moment. : 

Also, if we do #1 it would warn about unexpected stuff appearing in the YAML (e.g. type), which the current parser is not designed to do yet.

Reply:
Created https://github.com/COVESA/vehicle_service_catalog/issues/30 for the property discussion